eq-author-api/middleware/upsertUser.js

```diff
@@ -0,0 +1,21 @@
+const { pick, isEqual } = require("lodash/fp");
+
+const { getUserBySub, createUser, updateUser } = require("../utils/datastore");
+
+const checkForUpdates = (user, existingUser) => {
+  const pickRequiredFields = pick(["email", "name", "sub", "picture"]);
+  return isEqual(pickRequiredFields(user), pickRequiredFields(existingUser));
+};
+
+module.exports = async (req, res) => {
+  let user = await getUserBySub(req.auth.sub);
+
+  if (!user) {
+    user = await createUser(req.auth);
+  } else if (!checkForUpdates(user, req.auth)) {
+    Object.assign(user, req.auth);
+    user = await updateUser(user);
+  }
+  res.send(200);
+  return user;
```

Comment:
I don't think we should be returning this.
```suggestion
  return;
```